src/libxtp/gwbse/bse.cc

```diff
@@ -409,18 +394,24 @@ template <typename T,int factor>
     BSE::Interaction BSE::Analyze_eh_interaction(const QMStateType& type) {
 
       Interaction analysis;
-      MatrixXfd H = MatrixXfd::Zero(_bse_size, _bse_size);
-      Add_Hqp<real_gwbse>(H);
+      CTP_LOG(ctp::logDEBUG, _log) << ctp::TimeStamp() << " ANALYZE_EH_INTERACTION DISABLED" << flush;
+
+      BSE_OPERATOR h (_orbitals, _log, _Mmn, _Hqp);
```

Comment:
I think it should work almost out of the box since we already have the Operator * Vector defined for the matrix free operators. However you also have operations like Vector * Operator involved and we should rewrite them as Operator*Vector and be careful

Reply:
all of these operations take expectation values so vector.T*operator*vector, so if we divide it like vector.T*(operator*vector),  we are fine

Reply:
I think the vector.T * Operator didn't work, but I might be wrong. I'll test

Reply:
yeah should work indeed

Reply:
Yup that's working